frameworks/MLPlanSKLearn/exec.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import math
+import os
+
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+from amlb.datautils import reorder_dataset
+from amlb.results import NoResultError, save_predictions_to_file
+from amlb.utils import dir_of, path_from_split, run_cmd, split_path, Timer
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset: Dataset, config: TaskConfig):
+    log.info("\n**** ML-Plan for scikit-learn ****\n")
+
+    # Mapping of benchmark metrics to Weka metrics
+    metrics_mapping = dict(
+        acc='ERRORRATE',
+        auc='AUC',
+        logloss='LOGLOSS',
+	f1='F1',
+	r2='R2',
+	rmse='ROOT_MEAN_SQUARED_ERROR',
+	mse='MEAN_SQUARED_ERROR',
+	rmsle='ROOT_MEAN_SQUARED_LOGARITHM_ERROR',
+	mae='MEAN_ABSOLUTE_ERROR'
+    )
+    metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+    if metric is None:
+	    raise ValueError('Performance metric {} is not supported.'.format(config.metric))
+
+    train_file = dataset.train.path
+    test_file = dataset.test.path
+    # Weka requires target as the last attribute
+    if dataset.target.index != len(dataset.predictors):
+        train_file = reorder_dataset(dataset.train.path, target_src=dataset.target.index)
+        test_file = reorder_dataset(dataset.test.path, target_src=dataset.target.index)
+
+    training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
+
+    cmd_root = "java -jar {here}/lib/mlplan/mlplan-cli*.jar ".format(here=dir_of(__file__))
+    cmd_params = dict(
+        f='"{}"'.format(train_file),
+        p='"{}"'.format(test_file),
+        t=config.max_runtime_seconds,
+        ncpus=config.cores,
+        l=metric,
+	    s=config.seed,   # weka accepts only int16 as seeds
+	    ooab=config.output_predictions_file,
```

Comment:
is this predictions file already in the expected format?
Usually, we ask the frameworks to provide the predictions and probabilities + various flags to generate the predictions file in a standard way. 
see for example:
```python
    return result(output_file=config.output_predictions_file,
                  predictions=predictions,
                  truth=y_test,
                  probabilities=probabilities,
                  probabilities_labels=prob_labels,
                  target_is_encoded=False,
                  models_count=num_models_trained,
                  models_ensemble_count=num_models_ensemble,
                  training_duration=training.duration,
                  predict_duration=predict.duration)
```
If you provide `probabilities_labels` like in the snippet above, the probabilities columns will be automatically reordered to meet sklearn expectations when computing metrics like AUC.

Reply:
I extended the CLI of ML-Plan to conform the output format as required by the benchmark which will produce a csv file where the first columns contain the probabilities of the respective class labels in case of classification, the predictions, and the truth (the latter two are written out for both tasks classification and regression). However, there was no information about the number of models trained etc. I am currently extending the output so that we can use the result function.

Reply:
great, let me verify that you can call the `result` function even with `predictions=None` for example, otherwise I'll fix it.

Reply:
You dont need to check that. I already added another output format and called the result in the ordinary way. So no worries.